x-pack/legacy/plugins/apm/server/lib/settings/agent_configuration/search.ts

```diff
@@ -27,26 +23,28 @@ export async function searchConfigurations({
 }) {
   const { client, config } = setup;
 
-  const filters: ESFilter[] = [{ term: { [SERVICE_NAME]: serviceName } }];
-
-  if (environment) {
-    filters.push({ term: { [SERVICE_ENVIRONMENT]: environment } });
-  } else {
-    filters.push({
-      bool: { must_not: { exists: { field: SERVICE_ENVIRONMENT } } }
-    });
-  }
+  const environmentFilter = environment
+    ? [{ term: { [SERVICE_ENVIRONMENT]: { boost: 2, value: environment } } }]
+    : [];
 
   const params = {
     index: config.get<string>('apm_oss.apmAgentConfigurationIndex'),
     body: {
       size: 1,
       query: {
-        bool: { filter: filters }
+        bool: {
```

Comment:
Maybe a comment here to clarify the sorting algorithm?

Reply:
Done